File: cpp/Vetores/aprovados.cpp
#include <iostream>
using namespace std;

int main() {
    int N;

    cout << "Quantos alunos serao digitados? ";
    cin >> N;

    string nomes[N];
    double nota1[N], nota2[N];

    for (int i = 0; i < N; i++) {
        cout << "Dados do " << i + 1 << "o aluno:" << endl;

        cout << "Nome: ";
        cin >> nomes[i];

        cout << "Nota do primeiro semestre: ";
        cin >> nota1[i];

        cout << "Nota do segundo semestre: ";
        cin >> nota2[i];
    }

    cout << "\nALUNOS APROVADOS:" << endl;

    for (int i = 0; i < N; i++) {
        double media = (nota1[i] + nota2[i]) / 2.0;

        if (media >= 6.0) {
            cout << nomes[i] << endl;
        }
    }

    return 0;
}

File: java/Vetores/Aprovados.java
import java.util.Locale;
import java.util.Scanner;

public class Aprovados {
    public static void main(String[] args) {

        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        int N;

        System.out.print("Quantos alunos serao digitados: ");
        N = sc.nextInt();

        String[] nomes = new String[N];
        double[] nota1 = new double[N];
        double[] nota2 = new double[N];

        for (int i = 0; i < N; i++) {

            System.out.println("Digite os dados do " + (i + 1) + "º aluno:");

            System.out.print("Nome: ");
            nomes[i] = sc.next();

            System.out.print("Nota do primeiro semestre: ");
            nota1[i] = sc.nextDouble();

            System.out.print("Nota do segundo semestre: ");
            nota2[i] = sc.nextDouble();
        }

        System.out.println("\nAlunos aprovados:");

        for (int i = 0; i < N; i++) {

            double media = (nota1[i] + nota2[i]) / 2.0;

            if (media >= 6.0) {
                System.out.println(nomes[i]);
            }
        }

        sc.close();
    }
}